refactor: CircularIconBadge에 클릭 가능 상태 추가 및 스타일 업데이트

icw0201 opened the pull request:
## 📌 Issue

<!-- 해결하려는 이슈 번호나 주제를 명확하게 적어주세요. -->

- 관련 이슈: close #177 

---

## 🛠 작업 내용

- CircularIconBadge에 클릭 가능 상태 추가 및 스타일 업데이트
- ***

## 🚀 기타 사항

<!-- 리뷰어가 추가적으로 알아야 할 사항이 있다면 기재해주세요. -->

- 추가적인 내용을 작성해주세요.(참고 자료, 협업 내용)

- [x] build 체크 했나요? : 넹 eslint 오류 나는데 이건 @layout-SY 님이 나중에 바꿔준대요

- [x] dev를 pull 받은 뒤 merge 했나요? : 최신 dev에서 분기했기에 현재 안해도 됩니다.

- 지금 PR 열어두고 api 연결 작업 중입니다!

icw0201 committed:
feat: CircularIconBadge에 클릭 가능 상태 추가 및 스타일 업데이트

aiden0413 commented on app/(anon)/test/style-badges/page.tsx:
```diff
@@ -1,8 +1,19 @@
+    alert(`unchecked 뱃지를 클릭했습니다! (${clickCount + 1}번째 클릭)`);
+    
+    // 여기에 API 호출 로직을 추가할 수 있습니다
+    // 예: fetch('/api/update-status', { method: 'POST' })
```
url에서 steps/4/3 stepNumber=4 detail=3 이걸로 step result upsert 구현하기

icw0201 replied:
@aiden0413 
- query 사용
- util 다른 곳에도 적용
- 또뭐더라? 했습니다

Jihye-kr reviewed (approved):
수고하셨습니다! 👍 👍 👍 👍 💯

icw0201 committed:
feat: circularIcom api 연결 WIP

icw0201 committed:
feat: circular 뱃지 api 연결 WIP

icw0201 committed:
Merge branch 'develop' into refactor/#177

icw0201 committed:
feat: circularIconBadge 컴포넌트 스타일 및 기능 개선
- 코드 정리
- api 연결 click 이벤트 구현 및 관련 front api 파일 생성성

icw0201 committed:
refactor: CircularIconBadge 컴포넌트 리팩토링 및 API 연결 개선
- useGetStepDetail 훅을 useGetStepResult로 변경하여 API 호출 방식 개선
- CircularIconBadge에서 'uncheck'를 'unchecked'로 수정하여 일관성 유지
- 로딩 상태 및 클릭 이벤트 처리 개선
- 관련 스타일 및 테스트 코드 업데이트

icw0201 committed:
Merge branch 'develop' into refactor/#177

icw0201 committed:
fix: eslint 경고 해결
